proto/connector/v1/connector.proto

```diff
@@ -8,220 +8,236 @@ import "google/protobuf/struct.proto";
-    // outstanding and should stop the stream only after all acknowledgments
-    // were received.
-    rpc Run(stream Source.Run.Request) returns (stream Source.Run.Response);
-    // Stop signals to the plugin to stop retrieving new records and flush any
-    // records that might be cached. The stream should will still remain open so
-    // Conduit can fetch the remaining records and send back any outstanding
-    // acknowledgments. After all acknowledgments are received the plugin should
-    // close the stream.
-    rpc Stop(Source.Stop.Request) returns (Source.Stop.Response);
-    // Teardown signals to the plugin that there will be no more calls to any
-    // other function. After Teardown returns, the plugin should be ready for a
-    // graceful shutdown.
-    rpc Teardown(Source.Teardown.Request) returns (Source.Teardown.Response);
+  // Configure is the first function to be called in a plugin. It provides the
+  // plugin with the configuration that needs to be validated and stored. In
+  // case the configuration is not valid it should return an error status.
+  rpc Configure(Source.Configure.Request) returns (Source.Configure.Response);
+  // Start is called after Configure to signal the plugin it can prepare to
+  // start producing records. If needed, the plugin should open connections in
+  // this function. The position parameter will contain the position of the
+  // last record that was successfully processed. The Source should therefore
+  // start producing records after this position.
+  rpc Start(Source.Start.Request) returns (Source.Start.Response);
+  // Run will open a bidirectional stream between Conduit and the plugin.
+  // The plugin is responsible for fetching records from 3rd party resources
+  // and sending them as responses to Conduit. Conduit will process the
+  // records asynchronously and send acknowledgments back to the plugin to
+  // signal that a record at a certain position was processed. Acknowledgments
+  // will be sent back to the connector in the same order as the records
+  // produced by the connector. If a record could not be processed by Conduit,
+  // the stream will be closed without an acknowledgment being sent back to the
+  // connector.
+  rpc Run(stream Source.Run.Request) returns (stream Source.Run.Response);
+  // Stop signals to the plugin to stop retrieving new records and flush any
+  // records that might be cached into the stream. It should block until it can
+  // determine the last record that will be sent to the stream and return the
+  // position of the last record. Conduit will keep the stream open until it
+  // receives the last record and sends back any outstanding acknowledgments.
+  // If Conduit did not send an acknowledgment for a record after the stream is
+  // closed, it should be interpreted as a negative acknowledgment.
+  rpc Stop(Source.Stop.Request) returns (Source.Stop.Response);
```

Comment:
This comment changed, the source should send back the last record sent to Conduit, this gives Conduit the ability to control the graceful stop.